lib/modules/manager/gleam/update-locked.ts

```diff
@@ -0,0 +1,25 @@
+import { logger } from '../../../logger';
+import type { UpdateLockedConfig, UpdateLockedResult } from '../types';
+import { extractLockFileContentVersions } from './locked-version';
+
+export function updateLockedDependency(
+  config: UpdateLockedConfig,
+): UpdateLockedResult {
+  const { depName, currentVersion, newVersion, lockFile, lockFileContent } =
+    config;
+  logger.debug(
+    `gleam.updateLockedDependency: ${depName}@${currentVersion} -> ${newVersion} [${lockFile}]`,
+  );
+  try {
+    const locked = extractLockFileContentVersions(lockFileContent!);
+    if (locked?.get(depName)?.find((version) => version === newVersion)) {
+      return { status: 'already-updated' };
+    }
+    // we don't currently support directly updating the lock file
+    // so we'll use gleam itself (via updateArtifacts()) to update the lock file instead
```

Comment:
do we need this function at all? isn't the artifacts function already handling it? 🤔

Reply:
It bothered me, having to restrict usage of the `update-lockfile` `rangeStrategy`, especially since it is common in Gleam package files to use broad version ranges, such as `>= 1.0.0 and < 2.0.0`. Even though this function itself does not do anything directly, there seemed to be some [precedent for this in the Cargo manager](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/cargo/update-locked.ts#L18).

From what I can tell, using `update-lockfile` strategy allows us to enter into [get-updated.ts#L131](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/update/branch/get-updated.ts#L131), and remember the `'unsupported'` update at [get-updated.ts#L147](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/update/branch/get-updated.ts#L147), so that ultimately we trigger `updateArtifacts()` to be run at [get-updated.ts#L339](https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/update/branch/get-updated.ts#L339). If this understand is correct, this would mean we do need `updateLockedDependency()` even if we do not yet support directly updating the lockfile.

Of note, this also lays the ground work to actually implement direct updates of the lockfile, but there are a few things that must be worked out on the Gleam-side first for that to happen. This should allow users to get most or all of the benefits today.

Reply:
if `unsupported` it's doing this
https://github.com/renovatebot/renovate/blob/302718f38e54c6c7590f488d749fb4da8362f36f/lib/workers/repository/update/branch/get-updated.ts#L144-L149

it doing the same if no `updateLockedDependency` 

https://github.com/renovatebot/renovate/blob/302718f38e54c6c7590f488d749fb4da8362f36f/lib/workers/repository/update/branch/get-updated.ts#L172-L181


So i still don't see any benefit to add this dummy function

Reply:
thank you, i see this now! removing & testing...